interfaces/builtin/desktop_legacy.go

```diff
@@ -241,6 +241,15 @@ dbus (send)
 # parallel-installs: this leaks read access to desktop files owned by keyed
 # instances of @{SNAP_NAME} to @{SNAP_NAME} snap
 /var/lib/snapd/desktop/applications/@{SNAP_INSTANCE_NAME}_*.desktop r,
+
+# glib-networking's GLib proxy (different than the portal's proxy service
+# org.freedesktop.portal.ProxyResolver)
+dbus (send)
+    bus=session
+    path=/org/gtk/GLib/PACRunner
+    interface=org.gtk.GLib.PACRunner
+    member=Lookup
+    peer=(label=unconfined),
```

Comment:
This seems fairly safe to add to `desktop-legacy`.  We probably want to direct people towards `org.freedesktop.portal.ProxyResolver` instead though.

If a confined app is hitting this D-Bus API, then it is using the libgiognomeproxy GProxyResolver backend, and has `gsettings` access to know that proxy auto config has been enabled.

If instead portal support is enabled, the portal backend for GProxyResolver is used.  This one doesn't need to have `gsettings` plugged, and delegates all proxy requests to xdg-desktop-portal, which in turn calls the standard GProxyResolver backend.

As far as security concerns over the `org.gtk.GLib.PACRunner` interface specifically, the client can ask the unconfined `glib-pacrunner` executable to execute an arbitrary `file://`, `http://`, or `https://` URL as JavaScript and pass in an arbitrary URL as a function argument.  With that said, the JS is run within mozjs's web sandbox so should be fairly safe.

Reply:
@kenvandine - fyi, this came up due to a snap-store bug: https://bugs.launchpad.net/snapd/+bug/1871364 and based on @jhenstridge comments, it seems like it needs to be update.